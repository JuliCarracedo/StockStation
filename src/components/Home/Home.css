.homeContainer {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.welcome-home {
  box-sizing: border-box;
  padding: 0 5%;
  height: 40%;
}

.links {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.filter-container {
  background-color: rgb(21, 82, 56);
  width: 100%;
  box-sizing: border-box;
  padding: 10px;
  display: flex;
  height: 2%;
}

.filter {
  width: 40%;
}

.company-link {
  display: flex;
  flex-direction: column;
  width: 50%;
  height: 20vh;
  align-items: flex-end;
  box-sizing: border-box;
  justify-content: flex-end;
  padding: 2%;
}

.company-link:nth-child(8) {
  background-color: rgb(28, 107, 73);
}

.company-link:nth-child(7) {
  background-color: rgb(21, 82, 56);
}

.company-link:nth-child(6) {
  background-color: rgb(21, 82, 56);
}

.company-link:nth-child(5) {
  background-color: rgb(28, 107, 73);
}

.company-link:nth-child(4) {
  background-color: rgb(28, 107, 73);
}

.company-link:nth-child(3) {
  background-color: rgb(21, 82, 56);
}

.company-link:nth-child(2) {
  background-color: rgb(21, 82, 56);
}

.company-link:nth-child(1) {
  background-color: rgb(28, 107, 73);
}

.fill {
  fill: white;
}

.company-title {
  margin: 6% 1% 1% 1%;
}

.unstyled {
  text-decoration: none;
  color: white;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}
