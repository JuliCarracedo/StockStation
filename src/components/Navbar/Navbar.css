.navbar {
  display: flex;
  box-sizing: border-box;
  width: 100%;
  padding: 1% 5%;
  justify-content: space-between;
  align-items: center;
  background-color: rgb(21, 82, 56);
}

.nav-links {
  width: 20%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.header-text {
  width: 45%;
  display: flex;
  justify-content: end;
  font-weight: 600;
  color: white;
}

.nav-link {
  fill: white;
}
