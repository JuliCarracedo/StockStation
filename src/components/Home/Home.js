import React from 'react';
import CompanyLink from '../CompanyLink';
import "./Home.css";

const Home = () => {
  const companies = ['AAPL', 'GGAL', 'FB', 'INTC', 'AMD', 'TSLA'];
  return (
    <div className="home-container">
      <div className="welcome-home">
        <h1>Stock Station</h1>
        <p>Check financial data for the biggest tech companies</p>
      </div>
        <select name="fitler" defaultValue="default">
          <option disabled value="default">Filter by</option>
          <option value="date">Date</option>
        </select>
      <div className="links">
        {companies.map((company) => <CompanyLink key={company} title={company} />)}
      </div>
    </div>
  );
};

export default Home;
