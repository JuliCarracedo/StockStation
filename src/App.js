/* eslint-disable no-nested-ternary */

import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import './App.css';
import CompanyPage from './components/CompanyPage';
import Home from './components/Home/Home';
import Navbar from './components/Navbar/Navbar';
import requestData, { companies } from './redux/stocks/async';

function App() {
  const { loading, error } = useSelector((store) => store.stocks);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(requestData());
  }, []);

  return (
    <div className="App">
      <BrowserRouter>
        <Navbar />
        {error ? (<div>API request quota reached</div>) : (loading ? <div> Loading</div> : (
          <Routes>
            <Route exact path="/" element={<Home />} />
            {companies.map((company) => <Route key={company} exact path={`/${company}`} element={<CompanyPage title={company} />} />)}
          </Routes>
        )) }
      </BrowserRouter>
    </div>
  );
}

export default App;
