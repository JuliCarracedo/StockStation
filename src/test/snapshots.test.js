import React from 'react';
import { Provider } from 'react-redux';
import renderer, { act } from 'react-test-renderer';
import { enableFetchMocks } from 'jest-fetch-mock';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/storeConfig';
import App from '../App';
import Navbar from '../components/Navbar/Navbar';
import Home from '../components/Home/Home';

enableFetchMocks();

describe('Application components', () => {
  beforeEach(() => {
    const list = [
      {
        date: '2020-12-31',
        symbol: 'A',
        revenue: 2,
        grossProfit: 2,
        ebitda: 3,
        netIncome: 4,
        eps: 5,
      }];
    fetch.mockResponse(JSON.stringify(list));
  });
  test('App component Snapshot', () => {
    let component;
    act(() => {
      component = renderer.create(<React.StrictMode>
        <Provider store={store}>
          <App />
        </Provider>
      </React.StrictMode>);
    });

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Navbar component Snapshot', () => {
    let component;
    act(() => {
      component = renderer.create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>,
      );
    });

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Home component Snapshot', () => {
    let component;
    act(() => {
      component = renderer.create(
        <Provider store={store}>
            <Home />
        </Provider>,
      );
    });

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
