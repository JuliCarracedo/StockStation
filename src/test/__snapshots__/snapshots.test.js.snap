// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application components App component Snapshot 1`] = `
<div
  className="App"
>
  <nav
    className="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        className="nav-link"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.67 0l2.83 2.829-9.339 9.175 9.339 9.167-2.83 2.829-12.17-11.996z"
        />
      </svg>
    </a>
    <div
      className="header-text"
    >
      <p>
        Stocks Station
      </p>
    </div>
    <div
      className="nav-links"
    >
      <svg
        className="nav-link"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 11c0 2.209-1.791 4-4 4s-4-1.791-4-4v-7c0-2.209 1.791-4 4-4s4 1.791 4 4v7zm4-2v2c0 4.418-3.582 8-8 8s-8-3.582-8-8v-2h2v2c0 3.309 2.691 6 6 6s6-2.691 6-6v-2h2zm-7 13v-2h-2v2h-4v2h10v-2h-4z"
        />
      </svg>
      <svg
        className="nav-link"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 13.616v-3.232c-1.651-.587-2.694-.752-3.219-2.019v-.001c-.527-1.271.1-2.134.847-3.707l-2.285-2.285c-1.561.742-2.433 1.375-3.707.847h-.001c-1.269-.526-1.435-1.576-2.019-3.219h-3.232c-.582 1.635-.749 2.692-2.019 3.219h-.001c-1.271.528-2.132-.098-3.707-.847l-2.285 2.285c.745 1.568 1.375 2.434.847 3.707-.527 1.271-1.584 1.438-3.219 2.02v3.232c1.632.58 2.692.749 3.219 2.019.53 1.282-.114 2.166-.847 3.707l2.285 2.286c1.562-.743 2.434-1.375 3.707-.847h.001c1.27.526 1.436 1.579 2.019 3.219h3.232c.582-1.636.75-2.69 2.027-3.222h.001c1.262-.524 2.12.101 3.698.851l2.285-2.286c-.744-1.563-1.375-2.433-.848-3.706.527-1.271 1.588-1.44 3.221-2.021zm-12 2.384c-2.209 0-4-1.791-4-4s1.791-4 4-4 4 1.791 4 4-1.791 4-4 4z"
        />
      </svg>
    </div>
  </nav>
  <div
    className="loading-container"
  >
    <h2
      className="loading"
    >
      LOADING
    </h2>
  </div>
</div>
`;

exports[`Application components Home component Snapshot 1`] = `
<div
  className="home-container"
>
  <div
    className="welcome-home"
  >
    <h1>
      Stock Station
    </h1>
    <p>
      Check financial data for the biggest tech companies
    </p>
  </div>
  <div
    className="filter-container"
  >
    <select
      defaultValue="default"
      name="fitler"
      onChange={[Function]}
    >
      <option
        disabled={true}
        value="default"
      >
        Filter by
      </option>
      <option
        value="date"
      >
        Date
      </option>
      <option
        value="income"
      >
        Income
      </option>
      <option
        value="alphabetical"
      >
        Alphabetical
      </option>
      <option
        value="eps"
      >
        EPS
      </option>
    </select>
  </div>
  <div
    className="links"
  />
</div>
`;

exports[`Application components Navbar component Snapshot 1`] = `
<nav
  className="navbar"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <svg
      className="nav-link"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16.67 0l2.83 2.829-9.339 9.175 9.339 9.167-2.83 2.829-12.17-11.996z"
      />
    </svg>
  </a>
  <div
    className="header-text"
  >
    <p>
      Stocks Station
    </p>
  </div>
  <div
    className="nav-links"
  >
    <svg
      className="nav-link"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 11c0 2.209-1.791 4-4 4s-4-1.791-4-4v-7c0-2.209 1.791-4 4-4s4 1.791 4 4v7zm4-2v2c0 4.418-3.582 8-8 8s-8-3.582-8-8v-2h2v2c0 3.309 2.691 6 6 6s6-2.691 6-6v-2h2zm-7 13v-2h-2v2h-4v2h10v-2h-4z"
      />
    </svg>
    <svg
      className="nav-link"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M24 13.616v-3.232c-1.651-.587-2.694-.752-3.219-2.019v-.001c-.527-1.271.1-2.134.847-3.707l-2.285-2.285c-1.561.742-2.433 1.375-3.707.847h-.001c-1.269-.526-1.435-1.576-2.019-3.219h-3.232c-.582 1.635-.749 2.692-2.019 3.219h-.001c-1.271.528-2.132-.098-3.707-.847l-2.285 2.285c.745 1.568 1.375 2.434.847 3.707-.527 1.271-1.584 1.438-3.219 2.02v3.232c1.632.58 2.692.749 3.219 2.019.53 1.282-.114 2.166-.847 3.707l2.285 2.286c1.562-.743 2.434-1.375 3.707-.847h.001c1.27.526 1.436 1.579 2.019 3.219h3.232c.582-1.636.75-2.69 2.027-3.222h.001c1.262-.524 2.12.101 3.698.851l2.285-2.286c-.744-1.563-1.375-2.433-.848-3.706.527-1.271 1.588-1.44 3.221-2.021zm-12 2.384c-2.209 0-4-1.791-4-4s1.791-4 4-4 4 1.791 4 4-1.791 4-4 4z"
      />
    </svg>
  </div>
</nav>
`;
